docs/applications/implementations/models.md

```diff
@@ -50,18 +50,50 @@ def create_estimator(run_config, model_config):
 
 You can import PyPI packages or your own Python packages to help create more complex models. See [Python Packages](../advanced/python-packages.md) for more details.
 
+The following packages have been pre-installed and can be used in your implementations:
+
+```text
+tensorflow==1.12.0
+boto3==1.9.78
+msgpack==0.6.1
+numpy>=1.13.3,<2
+requirements-parser==0.2.0
+packaging==19.0.0
+```
+
+You can install additional PyPI packages and import your own Python packages. See [Python Packages](../advanced/python-packages.md) for more details.
+
 
 # Tensorflow Transformations
-You can preprocess input features and labels to your model by defining a `transform_tensorflow` function. This is useful in cases where you don't have access to the model implementation which expects a multi-dimensional tensor, for example.
+You can preprocess input features and labels to your model by defining a `transform_tensorflow` function. An example of when this might be useful is to reshape a tensor to feed into a pre-made model.
 
 ```python
 def transform_tensorflow(features, labels, model_config):
+     """Create a mapping function for the tf.Dataset API. You can define
+     tensor transformations you want to apply to the features and labels of
+     each training sample.
```

Comment:
Could "each training sample" be misleading? Is the tensor that is passed into this function a single sample or multiple? Also the word "training" is unnecessary since this is also applied at prediction time